src/components/Graph.react.js

```diff
@@ -73,208 +54,92 @@ const filterEventData = (gd, eventData, event) => {
 class PlotlyGraph extends Component {
     constructor(props) {
         super(props);
-        this.gd = React.createRef();
-        this.bindEvents = this.bindEvents.bind(this);
-        this._hasPlotted = false;
-        this._prevGd = null;
-        this.graphResize = this.graphResize.bind(this);
+
+        this.state = {
+            extendData: [],
+        };
+
+        this.clearExtendData = this.clearExtendData.bind(this);
     }
 
-    plot(props) {
-        const {figure, animate, animation_options, config} = props;
-        const gd = this.gd.current;
-        if (
-            animate &&
-            this._hasPlotted &&
-            figure.data.length === gd.data.length
-        ) {
-            return Plotly.animate(gd, figure, animation_options);
+    componentDidMount() {
+        if (this.props.extendData) {
+            this.setState({
+                extendData: [this.props.extendData],
+            });
```

Comment:
`extendData` does not respect the usual pattern of props for Dash - as the wrapper waits for the real component to be loaded, it accumulates updates to extendData, the real component is now capable of processing an array of `extendData` instead of an object. The array is cleared upon processing.